pallets/automation-time/src/tests.rs

```diff
@@ -311,6 +340,70 @@ fn cancel_works_for_scheduled() {
 	})
 }
 
+#[test]
+fn cancel_works_for_multiple_executions_scheduled() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let owner: AccountId = ALICE;
+		let task_id1 = schedule_task(
+			owner,
+			vec![40],
+			vec![SCHEDULED_TIME, SCHEDULED_TIME + 120, SCHEDULED_TIME + 240],
+			vec![2, 4, 5],
+		);
+		LastTimeSlot::<Test>::put((SCHEDULED_TIME - 420, SCHEDULED_TIME - 420));
+		System::reset_events();
+
+		match AutomationTime::get_scheduled_tasks(SCHEDULED_TIME) {
+			None => {
+				panic!("A task should be scheduled")
+			},
+			Some(task_ids) => {
+				assert_eq!(task_ids.len(), 1);
+				assert_eq!(task_ids[0], task_id1);
+			},
+		}
+		match AutomationTime::get_scheduled_tasks(SCHEDULED_TIME + 120) {
+			None => {
+				panic!("A task should be scheduled")
+			},
+			Some(task_ids) => {
+				assert_eq!(task_ids.len(), 1);
+				assert_eq!(task_ids[0], task_id1);
+			},
+		}
+		match AutomationTime::get_scheduled_tasks(SCHEDULED_TIME + 240) {
+			None => {
+				panic!("A task should be scheduled")
+			},
+			Some(task_ids) => {
+				assert_eq!(task_ids.len(), 1);
+				assert_eq!(task_ids[0], task_id1);
+			},
+		}
+
+		assert_ok!(AutomationTime::cancel_task(Origin::signed(owner), task_id1,));
+
+		if let Some(_) = AutomationTime::get_scheduled_tasks(SCHEDULED_TIME) {
+			panic!("Tasks scheduled for the time it should have been deleted")
+		}
+		if let Some(_) = AutomationTime::get_scheduled_tasks(SCHEDULED_TIME + 120) {
+			panic!("Tasks scheduled for the time it should have been deleted")
+		}
+		if let Some(_) = AutomationTime::get_scheduled_tasks(SCHEDULED_TIME + 240) {
```

Comment:
prob also worth checking the tasks map to make sure that's not still in there as well